src/test/java/com/exactpro/epfast/decoder/ascii/TestDecodeAsciiString.java

```diff
@@ -10,164 +12,254 @@
+                }
+                stringBuilder.append("c3");
 
-    @Test
-    void testSimpleStringGetValueTwice() throws IOException {
-        withByteBuf("41 42 c3", buffers -> {
-            decode(nullableStringDecoder, buffers);
-            assertTrue(nullableStringDecoder.isReady());
-            assertEquals("ABC", nullableStringDecoder.getValue());
-            assertEquals("ABC", nullableStringDecoder.getValue());
-        });
-    }
+                withByteBuf(stringBuilder.toString(), buffers -> {
+                    decode(decoder, buffers);
+                    assertTrue(decoder.isReady());
+                    assertThrows(OverflowException.class, () -> decoder.getValue());
+                });
+            }
+        }
 
-    @Test
-    void testMandatoryEmptyStringGetValueTwice() throws IOException {
-        withByteBuf("80", buffers -> {
-            decode(mandatoryStringDecoder, buffers);
-            assertTrue(mandatoryStringDecoder.isReady());
-            assertEquals("", mandatoryStringDecoder.getValue());
-            assertEquals("", mandatoryStringDecoder.getValue());
-        });
-    }
+        @Test
+        void testNullableOverlong() throws IOException {
+            withByteBuf("00 00 00 81", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                decoder.getValue();
+            });
+        }
+
+        @Test
+        void testNullableMaxAllowedLength() throws IOException {
+            StringBuilder stringBuilder = new StringBuilder(DecodeAsciiString.MAX_ALLOWED_LENGTH);
+            for (int i = 0; i < DecodeAsciiString.MAX_ALLOWED_LENGTH - 2; i += 2) {
+                stringBuilder.append("41 41 ");
+            }
+            stringBuilder.append("c3");
+
+            withByteBuf(stringBuilder.toString(), buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                assertEquals(DecodeAsciiString.MAX_ALLOWED_LENGTH - 1, decoder.getValue().length());
+            });
+        }
+
+        @Test
+        void testZeroByteStringNullableTwoValuesInRow() throws IOException {
+            withByteBuf("00 00 80 00 00 00 00 80", buffers -> {
+                decode(decoder, buffers);
+                assertTrue(decoder.isReady());
+                assertEquals("\0", decoder.getValue());
 
-    @Test
-    void testZeroByteStringNullableTwoValuesInRow() throws IOException {
-        withByteBuf("00 00 80 00 00 00 00 80", buffers -> {
-            decode(nullableStringDecoder, buffers);
-            assertTrue(nullableStringDecoder.isReady());
-            assertEquals("\0", nullableStringDecoder.getValue());
-
-            decode(nullableStringDecoder, buffers);
-            assertTrue(nullableStringDecoder.isReady());
-            assertEquals("\0\0\0", nullableStringDecoder.getValue());
-        });
+                decode(decoder, buffers);
```

Comment:
This is risky. We need to reconsider how to write such kind of tests. Our decoders have limitations, that they can't handle empty buffers. Here we may pass empty buffer (if the first decode fully consumes one of the buffers). I think we need to return a list of remaining buffers from decode() and pass this potentially smaller set to the next decode()